registry.ts

```diff
@@ -154,6 +154,76 @@ async function unpkgVersions(name: string): Promise<string[]> {
   return m.map((x) => x[1]);
 }
 
+interface PackageInfo {
+  parts: string[];
+  scope: string;
+  packageName: string;
+  version: string;
+}
+function defaultParts(that: RegistryUrl): PackageInfo {
+  const parts = that.url.split("/");
+  const [packageName, version] = parts[4].split("@");
+  if (parts[3] === undefined) {
+    throw new Error(`Package scope not found in ${that.url}`);
+  }
+  if (packageName === undefined) {
+    throw new Error(`Package name not found in ${that.url}`);
+  }
+  if (version === undefined) {
+    throw new Error(`Unable to find version in ${that.url}`);
+  }
+  return {
+    scope: parts[3],
+    packageName,
+    version,
+    parts,
+  };
+}
+function defaultScopeAt(that: RegistryUrl, version: string): string {
+  const { parts, packageName } = defaultParts(that);
+  parts[4] = `${packageName}@${version}`;
+  return parts.join("/");
+}
+export function _defaultAt(
+  that: RegistryUrl,
+  version: string,
+  hasScope = false,
+): string {
+  if (!hasScope) return that.url.replace(/@(.*?)(\/|$)/, `@${version}/`);
```

Comment:
This is probably too clever :), can you delete _defaultAt  (unused I think) ?

Reply:
Oops! I forgot deleting this. Thanks!
(I wonder if there is such a lint feature which tells me unused export functions and values...)